media_driver/linux/common/codec/ddi/media_ddi_decode_base.cpp

```diff
@@ -1096,13 +1096,10 @@ VAStatus DdiMediaDecode::CreateBuffer(
 
     if( true == buf->bCFlushReq )
     {
-        mos_bo_subdata(buf->bo, buf->uiOffset, size * numElements, data);
-    }
-    else
-    {
-        status = MOS_SecureMemcpy((void *)(buf->pData + buf->uiOffset), size * numElements, data, size * numElements);
-        DDI_CHK_CONDITION((status != MOS_STATUS_SUCCESS), "DDI:Failed to copy buffer data!", VA_STATUS_ERROR_OPERATION_FAILED);
+        mos_bo_wait_rendering(buf->bo);
     }
+    status = MOS_SecureMemcpy((void *)(buf->pData + buf->uiOffset), size * numElements, data, size * numElements);
```

Comment:
a bo_map before memory copy?

Reply:
I can't tell what the right thing is to do here.  If it really is as simple as dropping in a `mos_bo_map()` right after the `mos_bo_wait_rendering()`, I can do that.  That said, I think this one is actually ok too.  It's guarded by `if(true == buf->bCFlushReq)` and the one place that ever sets `bCFlushReq = true` does so right after setting `pData`.

That said, I'm happy to do whatever you want.  I'm coding entirely blind here; I've never touched the media driver before so I don't know any of your conventions.

Reply:
CFlushReq should be used for the sku which has no LLC, but in this case, they used to distinguish whether the bistream buffer is swapped into system memory or gpu memory.  and yes, it is already mapped.

Reply:
looks good to me.